joplin/base/wagtail_hooks.py

```diff
@@ -300,6 +301,158 @@ def register_link_handler(features):
     features.register_link_type(InternalLinkHandler)
 
 
+# In here we're going to remake all the urls for this page
+@hooks.register('after_edit_page')
+def after_edit_page(request, page):
+    # Make sure we're publishing so our Janis urls reflect the live site
+    # taking this from https://github.com/wagtail/wagtail/blob/349fca66f32ecf3df6316b0cffb4e5fd0d24d6c7/wagtail/admin/views/pages.py#L228
+    publishing = bool(request.POST.get('action-publish'))
+    if not publishing:
+        return
+
+    # Clear out all old urls
+    for janis_url in page.janis_urls.all():
```

Comment:
this seems wasteful. if the IA isn't changing, we shouldn't need to re-create the urls. 

I'm not 100% sure if it would be available in this hook, but as part of the form validation we get a list of changed fields. If the fields aren't related to IA, we could/should skip all this, no?

Reply:
Good ideas. This could balloon, so I made an issue for it https://github.com/cityofaustin/techstack/issues/3786